code/ed25519/Hacl.Impl.SHA512.ModQ.fst

```diff
@@ -62,9 +64,12 @@ let sha512_pre_pre2_msg hash prefix prefix2 len input =
   push_frame ();
   let h0 = ST.get () in
   let st = Hash.alloca_512 () in
-  let _ = Hash.update_512 st prefix 32ul in
-  let _ = Hash.update_512 st prefix2 32ul in
-  let _ = Hash.update_512 st input len in
+  let err0 = Hash.update_512 st prefix 32ul in
+  let err1 = Hash.update_512 st prefix2 32ul in
+  let err2 = Hash.update_512 st input len in
+  LowStar.Ignore.ignore err0; // Why do we ignore this error?
+  LowStar.Ignore.ignore err1; // Why do we ignore this error?
+  LowStar.Ignore.ignore err2; // Why do we ignore this error?
```

Comment:
because the post-condition is crafted in such a way that you can eliminate the error statically -- here, you know you won't exceed the maximum length, so the function will never error out. (In essence, the error condition is 100% characterized by the post-condition so that it can be statically eliminated via a case analysis.)